app/assets/javascripts/discourse-common/addon/resolver.js

```diff
@@ -85,252 +66,255 @@ export function buildResolver(baseName) {
         return Ember.TEMPLATES["admin/templates/admin"];
       }
 
-      if (
-        decamelized.indexOf("admin") === 0 ||
-        decamelized.indexOf("javascripts/admin") === 0
+      let namespaced, match;
+
+      if (parsedName.fullNameWithoutType.startsWith("components/")) {
+        // Look up components as-is
+        namespaced = parsedName.fullNameWithoutType;
+      } else if (/^admin[_\.-]/.test(parsedName.fullNameWithoutType)) {
+        namespaced = parsedName.fullNameWithoutType.slice(6);
+      } else if (
+        (match = parsedName.fullNameWithoutType.match(/^admin([A-Z])(.+)$/))
       ) {
-        decamelized = decamelized.replace(/^admin\_/, "admin/templates/");
-        decamelized = decamelized.replace(/^admin\./, "admin/templates/");
-        decamelized = decamelized.replace(/\./g, "_");
+        namespaced = `${match[1].toLowerCase()}${match[2]}`;
+      }
 
-        const dashed = decamelized.replace(/_/g, "-");
+      if (namespaced) {
+        let adminParsedName = this.parseName(`template:${namespaced}`);
         return (
-          Ember.TEMPLATES[decamelized] ||
-          Ember.TEMPLATES[dashed] ||
-          Ember.TEMPLATES[dashed.replace("admin-", "admin/")]
+          // Built-in
+          this.findTemplate(adminParsedName, "admin/templates/") ||
+          // Plugin
+          this.findTemplate(adminParsedName, "javascripts/admin/")
         );
       }
-    },
+    }
 
     findWizardTemplate(parsedName) {
-      let decamelized = decamelize(parsedName.fullNameWithoutType);
-      if (decamelized.startsWith("components")) {
-        let comPath = `wizard/templates/${decamelized}`;
-        const compTemplate =
-          Ember.TEMPLATES[`javascripts/${comPath}`] || Ember.TEMPLATES[comPath];
-        if (compTemplate) {
-          return compTemplate;
-        }
-      }
-
-      if (decamelized === "javascripts/wizard") {
+      if (parsedName.fullNameWithoutType === "wizard") {
         return Ember.TEMPLATES["wizard/templates/wizard"];
       }
 
-      if (
-        decamelized.startsWith("wizard") ||
-        decamelized.startsWith("javascripts/wizard")
-      ) {
-        decamelized = decamelized.replace(/^wizard\_/, "wizard/templates/");
-        decamelized = decamelized.replace(/^wizard\./, "wizard/templates/");
-        decamelized = decamelized.replace(/\./g, "_");
+      let namespaced;
 
-        const dashed = decamelized.replace(/_/g, "-");
-        return (
-          Ember.TEMPLATES[decamelized] ||
-          Ember.TEMPLATES[dashed] ||
-          Ember.TEMPLATES[dashed.replace("wizard-", "wizard/")]
+      if (parsedName.fullNameWithoutType.startsWith("components/")) {
+        // Look up components as-is
+        namespaced = parsedName.fullNameWithoutType;
+      } else if (/^wizard[_\.-]/.test(parsedName.fullNameWithoutType)) {
+        // This may only get hit for the loading routes and may be removable.
+        namespaced = parsedName.fullNameWithoutType.slice(7);
```

Comment:
Can we check this, and remove if it's not needed?

Reply:
We should probably investigate this further. I suggest doing it as a follow-up to this PR as we work to standardize things further.